pipeline/pipeline.go

```diff
@@ -57,7 +57,12 @@ type Step struct {
 }
 
 func RunPipeline(obj Object, data *Context, steps []Step) Result {
-	for _, step := range steps {
+	l := data.Log.V(2).WithName("RunPipeline")
+	l.Info("running steps", "steps", stepNames(steps))
+
+	for i, step := range steps {
+		r := step.F(obj, data)
+		l.Info("ran step", "step", step.Name, "result", r, "step_index", i)
 		if r := step.F(obj, data); r.Abort || r.Err != nil {
 			if r.Err == nil {
```

Comment:
I think we are now running the step twice?

Reply:
thanks for catching this one 🤦‍♀️  looks like coffee levels are not high enough today

Reply:
That's what code reviews are for. I would have hopped that tests would have caught that..